client/state/jetpack/credentials/actions.js

```diff
@@ -48,3 +52,29 @@ export const deleteCredentials = ( siteId, role ) => ( dispatch ) => {
 		role,
 	} );
 };
+
+export const testCredentials = ( siteId, role ) => ( {
+	type: JETPACK_CREDENTIALS_TEST,
+	siteId,
+	role,
+} );
+
+export const markCredentialsAsValid = ( siteId, role ) => ( {
+	type: JETPACK_CREDENTIALS_TEST_VALID,
+	siteId,
+	role,
+} );
+
+export const markCredentialsAsInvalid = ( siteId, role ) => ( {
+	type: JETPACK_CREDENTIALS_TEST_INVALID,
+	siteId,
+	role,
+} );
+
+export const updateCredentialsTestResult = ( siteId, role, testResult ) => ( dispatch ) => {
```

Comment:
If we use this thunk just once, I think it's an unnecessary indirection and can be inlined directly.

Reply:
Thanks for the suggestion. I just removed this function. In case we will need it in the future for multiple scenarios, then we will add it back.